### PR TITLE
Fix https://github.com/TypistTech/imposter-plugin/issues/43 array_merge 0 args given

### PR DESCRIPTION
**Pull request by cambell-prince**
See TypistTech/imposter-plugin#43

- Fix ArrayUtil::flatten for null or empty arrays

### CODE REVIEW

File: src/ArrayUtil.php [hunk 1]
```diff
@@ -21,6 +21,12 @@ public static function flattenMap(callable $callable, array $array): array
      */
     public static function flatten(array $array): array
     {
+        if ($array === null) {
```
Comment (TangRufus) [inline src/ArrayUtil.php]: `$array` will never be `null` because of parameter type hint on line 22.

**Comment by TangRufus** [comment]
Do you think we need a more robust implementation like: https://github.com/illuminate/support/blob/fe658b6365fdd2d9cfe50c207f262d93bf3cfd3d/Arr.php#L200-L224

File: src/ArrayUtil.php (hunk 1 above)
Comment (cambell-prince) [reply src/ArrayUtil.php]: Agreed.

**Comment by cambell-prince** [comment]
That implementation seems better.  The current implementation isn't recursive and wouldn't flatten an arbitrary n-dimensional array.

**Comment by TangRufus** [comment]
Should be fixed in https://github.com/TypistTech/imposter/pull/85
Thanks for the PR

**Pull request closed without merging**